x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -109,14 +136,43 @@ export const CreateAnalyticsForm: FC<CreateAnalyticsFormProps> = ({ actions, sta
         });
       }
     } catch (e) {
-      // TODO: ensure error messages show up correctly
       setFormState({ loadingDepFieldOptions: false, dependentVariableFetchFail: true });
     }
   };
 
+  const getSourceIndexErrorMessages = () => {
+    const errors = [];
+    if (!sourceIndexNameEmpty && !sourceIndexNameValid) {
+      errors.push(
+        <Fragment>
+          {i18n.translate('xpack.ml.dataframe.analytics.create.sourceIndexInvalidError', {
```

Comment:
As mentioned in kibana [best practices](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#best-practices), you should use `<FormattedMessage>` most of the time

Reply:
Good catch. Updated in b2e501c17ea72ad79e742cba55bb69bf75417f44